### DISCUSSION TITLE
electron10

### DISCUSSION THREAD

**Discussion by grahamperrin** [helloSystem/hello#223] [Q&A] [answered]
Spun off from https://github.com/helloSystem/hello/issues/161#issuecomment-946900935

> … Previously installed `electron10` …

@probonopd please, what was the origin? 

<https://www.freshports.org/search.php?stype=name&method=match&query=electron&num=20&deleted=includedeleted&orderby=port&orderbyupdown=asc&search=Search&format=html&branch=head> there's no _electron10_. 

https://cgit.freebsd.org/ports/log/?qt=grep&q=electron10 finds nothing. 

@probonopd what's removable?

`pkg autoremove --dry-run`

The full list, please.

**Comment by probonopd** [comment]
Before I remove anything:
Which Elecron is available for 12.2? Because currently:

```
FreeBSD% pkg search electron
FreeBSD% 
```

I am really worried that I am going to lose the remainders of Electron if I uninstall packages.

> @probonopd please, what was the origin?

```
FreeBSD% pkg info electron10
electron10-10.4.7
Name           : electron10
Version        : 10.4.7
Installed on   : Wed Oct 13 19:14:22 2021 CEST
Origin         : devel/electron10
```

So it seems like I installed it not _that_ long ago.

**Reply by grahamperrin** [reply]
> Before I remove anything:

A dry run removes nothing. Can you share the list, please?

> Which Elecron is available for 12.2?

[devel/electron11 and devel/electron12 · Discussion #224 · helloSystem/hello](https://github.com/helloSystem/hello/discussions/224)

**Comment by grahamperrin** [comment]
> … what was the origin? …

Sorry, I meant plain English origin, as in, where did it originate. The repository. 

@probonopd please run: 

`pkg query '%o %v %R' electron10`

**Reply by probonopd** [reply]
```
FreeBSD% pkg query '%o %v %R' electron10
devel/electron10 10.4.7 unknown-repository
```

Unfortunately I don't remember where I got it from, but it did work before the update of FreeBSD 12.2 and its packages.

**Reply by grahamperrin** [reply]
_unknown-repository_ may be seen for installations that result from `make install`. 

> … I don't remember …

Maybe you got it here: 

https://github.com/tagattie/FreeBSD-Electron/releases

@tagattie

**Reply by probonopd** [reply]
Rather unlikely, I'd remember that. But thanks for the link. At least I now have a place to get it _again_, should I ever need :-)

Very cool - that repository also has other versions.
I will see what I can do with them.

**Comment by grahamperrin** [comment, marked as answer]
Without knowing the source of devel/electron10: https://github.com/helloSystem/hello/discussions/223#discussioncomment-1525770 is closest to an answer, but GitHub doesn't allow me to mark it as such.
